app/models/problem_check.rb

```diff
@@ -70,14 +70,14 @@ def call
 
   private
 
-  def problem(override_key = nil)
+  def problem(override_key = nil, override_data = {})
```

Comment:
With this `override_data` and `override_key` stuff, I'm just wondering why in the problem check subclasses we didn't just do conditionals? For example in `SidekiqCheck`:

```
def translation_data
  if massive_queue?
    return { queue_size: Jobs.queued }
  end

  {}
```

I guess the main downside is having to do the check a second time. Not a huge deal, just wondering what your thoughts are?

Reply:
The intent is to have `translation_key` and `translation_data` be "dumb", almost like attributes, serving the happy path. I have an incoming PR where the translation key is looked up by it's conventional place, which will eliminate it from most checks. I'd be hesitant to couple them both to the implementation details of the check for the special cases.

On the other hand, now that you mention it, for `translation_data`, there's not really anything harmful about shipping a few extra keys that aren't needed for the different problems.

E.g. say case A requires key X, and case be requires key X and Y, we might just do:

```ruby
def translation_data = { x: "foo", y: "bar" }
```

without any need for a conditional.

If the two cases diverge too much, perhaps they should be split into two separate problem checks.

Reply:
Okay sounds reasonable :+1: I think it's fine for `translation_data` to ship extra unnecessary keys, they just get thrown away by I18n.t